magento/pwa-studio
# PWA-973: Use jsx-no-literals rule

## Description

Do not merge until PWA-1033 is merged.

This PR 
* adds the jsx-no-literals eslint rule with a configuration that meets our needs. 
* Updates eslint-plugin-react to 7.21.5 (latest).

It will prevent direct string usage in jsx but allow strings to be passed as `id`s. It cannot catch strings passed as `const`s but that's not something we should/can handle anyways

The following will error:

```
<div>{'oops!'}</div>
```

The following will not error:

```
const noError = 'No error!';
<div>{noError}</div>
```

```
<FormattedMessage id={'im.allowed'} />
```

## Related Issue





Closes [PWA-973](https://jira.corp.magento.com/browse/PWA-973).

## Acceptance 



### Verification Stakeholders

### Specification



### Verification Steps


1. Check out PR.
2. `yarn lint` should not error
3. Add a string literal to some jsx in a component, like `<span>{'literal'}</span>`.
4. `yarn lint` should error

## Screenshots / Screen Captures (if appropriate)

## Checklist

* I have updated the documentation accordingly, if necessary.
